bitbots_localization/src/localization.cpp

```diff
@@ -347,51 +347,28 @@ void Localization::getMotion() {
   geometry_msgs::TransformStamped transformStampedPast;
   geometry_msgs::TransformStamped transformStampedNow;
 
-  movement2_.x = 0;
-  movement2_.y = 0;
-  movement2_.z = 0;
-
   try {
 
     transformStampedNow = tfBuffer.lookupTransform("odom", "base_footprint", now);
 
-    ros::Time past = transformStampedNow.header.stamp - ros::Duration(0.04);  //TODO better param
+    ros::Time past = transformStampedNow.header.stamp - ros::Duration(1 / config_.publishing_frequency);
 
     transformStampedPast = tfBuffer.lookupTransform("odom", "base_footprint", past);
 
-    geometry_msgs::Vector3 past_;
-    geometry_msgs::Vector3 now_;
-
-    past_.x = transformStampedPast.transform.translation.x;
-    past_.y = transformStampedPast.transform.translation.y;
-    past_.z = tf::getYaw(transformStampedPast.transform.rotation);
-
-    now_.x = transformStampedNow.transform.translation.x;
-    now_.y = transformStampedNow.transform.translation.y;
-    now_.z = tf::getYaw(transformStampedNow.transform.rotation);
-
-    movement_.x = atan2(now_.y - past_.y, now_.x - past_.x) - past_.z; //deltaRot1
-    movement_.y = hypot(past_.x - now_.x, past_.y - now_.y); //deltaTrans
-    movement_.z = now_.z - past_.z - movement_.x; // deltaRot2
-
-    //rotational movement
-    geometry_msgs::Vector3 rotation_;
-    geometry_msgs::Vector3 translation_;
-    rotation_.x = 0;
-    rotation_.y = 0;
-    rotation_.z = tf::getYaw(transformStampedNow.transform.rotation) -
-        tf::getYaw(transformStampedPast.transform.rotation);
-
     //linear movement
-    translation_.x = transformStampedNow.transform.translation.x - transformStampedPast.transform.translation.x;
-    translation_.y = transformStampedNow.transform.translation.y - transformStampedPast.transform.translation.y;
-    translation_.z =
-        transformStampedNow.transform.translation.z - transformStampedPast.transform.translation.z; // oder 0?
+    cartesian_movement_.x = transformStampedNow.transform.translation.x - transformStampedPast.transform.translation.x;
+    cartesian_movement_.y = transformStampedNow.transform.translation.y - transformStampedPast.transform.translation.y;
+    cartesian_movement_.z =
+        transformStampedNow.transform.translation.z - transformStampedPast.transform.translation.z;
```

Comment:
do we really want to consider z here? i mean... well...

Reply:
Maybe we have flying robots is the future. /s